pkg/component/component.go

```diff
@@ -265,6 +258,15 @@ func (c *Component) FromRequestContext(ctx context.Context) context.Context {
 	}
 }
 
+// FrequencyPlansStore returns a FrequencyPlans Store based on the component config and the context provided
+func (c *Component) FrequencyPlansStore(ctx context.Context) (*frequencyplans.Store, error) {
+	fpsFetcher, err := c.config.FrequencyPlansFetcher(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return frequencyplans.NewStore(fpsFetcher), nil
```

Comment:
Right now the HTTP fetcher itself does the caching, but I believe that it would be better if `component.HTTPClient` has an option that allows it to attach a cache to the transport, and the cache is global to the component. In that case, even though we're recreating everything, the transport cache is stable.

Reply:
From last change the `frequencyPlans` is now a unexported variable that still is inside the component but only accessible via the `component.FrequencyPlansStore` method. Basically like it was before, so the recreation of the cache isn't happening anymore but it will be something to look at in the TTI implementation.